src/arviz_stats/bayes_factor.py

```diff
@@ -0,0 +1,54 @@
+from .base.density import _DensityBase 
+import numpy as np 
+import warnings
+
+def bayes_factor(idata, var_name, ref_val=0, return_ref_vals=False, prior=None):
+    """
+    Approximated Bayes Factor for comparing hypotheses of two nested models, 
+    using KDE for density estimation.
+    Parameters
+    ----------
+    idata : InferenceData
+        Object containing posterior and prior data.
+    var_name : str
+        Name of the variable to test.
+    ref_val : int or float, default 0
+        Reference (point-null) value for Bayes factor estimation.
+    return_ref_vals : bool, default False
+        If True, also return the values of prior and posterior densities at the reference value.
+    Returns
+    -------
+    dict
+        A dictionary with Bayes Factor values: BF10 (H1/H0 ratio) and BF01 (H0/H1 ratio).
+    """
+    posterior = extract(idata, var_names=var_name).values
+    prior = extract(idata, var_names=var_name, group="prior").values
+
+    if not isinstance(ref_val, (int, float)):
+        raise ValueError("The reference value (ref_val) must be a numerical value (int or float).")
+
+    if ref_val > posterior.max() or ref_val < posterior.min():
+        warnings.warn(
+            "The reference value is outside the posterior range. "
+            "This results in infinite support for H1, which may overstate evidence."
+        )
+    if posterior.dtype.kind == "f":
+        density_instance = _DensityBase()
+        posterior_grid, posterior_pdf, _ = density_instance._kde(x=posterior, grid_len=512, circular=False)
+        prior_grid, prior_pdf, _ = density_instance._kde(x=prior, grid_len=512, circular=False)
```

Comment:
@OriolAbril what do you think is the best way to implement this function? In particular this part?